components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -192,6 +192,9 @@ open class GleanInternalAPI {
             GleanInternalMetrics.clientId.generateAndSet()
         }
 
+        GleanInternalMetrics.appBuild.set(BuildConfig.VERSION_CODE.toString())
+        GleanInternalMetrics.appDisplayVersion.set(BuildConfig.VERSION_NAME)
```

Comment:
This is referencing the `BuildConfig` class of the library. I have no clue why this class has those static fields since they have no meaning in a library - but they are not pointing to the code/name of the application.

In the library you'll have to query the package manager at runtime. See here:
https://github.com/mozilla-mobile/android-components/blob/master/components/service/telemetry/src/main/java/org/mozilla/telemetry/util/ContextUtils.java#L19-L26

Reply:
Good point. I changed that, this is ready for another look, @pocmo . For some reason, in tests, `versionName` is always null. Do you know how can I set a proper test value with Robolectirc?

Reply:
I was able to mock that in the tests.